### PR TITLE
Switch use RArchOp instead of RAnalOp, remove duplicated enums and st…

### PR DESCRIPTION
**Pull request by trufae**
…ructs ##arch

- [ ] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [book](https://github.com/radareorg/radare2book) (optional)

**Description**

### CODE REVIEW

File: libr/include/r_arch.h [hunk 1]
```diff
@@ -51,6 +51,38 @@ typedef struct r_arch_config_t {
 
 #define	R_ARCH_CONFIG_IS_BIG_ENDIAN(cfg_)	(((cfg_)->endian & R_SYS_ENDIAN_BIG) == R_SYS_ENDIAN_BIG)
 
+typedef struct r_arch_op_hint_t {
```
Comment (condret) [inline libr/include/r_arch.h:54]: Why have hints in arch, feels more like they rather belong to anal

Comment (trufae) [reply libr/include/r_arch.h:54]: because it needs to be tied to the ArchOp struct

Comment (trufae) [reply libr/include/r_arch.h:54]: archophints are just overrides of the archop details

Comment (condret) [reply libr/include/r_arch.h:54]: does it? as far as I've seen, it is only used in arm to set bitness. So I replaced hint with ut32 in archop

**Pull request closed without merging**